modules/vocabularies/src/main/frontend/src/vocabularyAction.jsx

```diff
@@ -74,6 +80,64 @@ const useStyles = makeStyles(theme => ({
 
 export default function VocabularyAction(props) {
   const classes = useStyles();
+  const [displayPopup, setDisplayPopup] = React.useState(false);
+  const [linkedQuestions, setLinkedQuestions] = useState([]);
+  const [questionnaires, setQuestionnaires] = useState([]);
+  const handleOpen = () => {fetchQuestionnaires();}
+  const handleClose = () => {setDisplayPopup(false);}
+  const handleUninstall = () => {setDisplayPopup(false); props.uninstall();}
+
+  let fetchQuestionnaires = () => {
+    if (questionnaires.length === 0) {
+      // Send a fetch request to determine the questionnaires available
+      fetch('/query?query=' + encodeURIComponent('select * from [lfs:Questionnaire]'))
```

Comment:
If you don't use a more complicated query, you can just get `/Questionnaires.paginate`.

By default, at most 10 questionnaires are returned, which right now is OK, but once we have more than that you may miss some usages of the vocabulary. You should request a high `limit`.

I can't remember, why don't you query just the questionnaires that actually use this vocabulary?

```suggestion
      const query = `select n.* from [lfs:Questionnaire] as n inner join [lfs:Question] as q on isdescendantnode(q, n) where q.sourceVocabulary='${props.acronym}'`;
      fetch(`/query?query=${encodeURIComponent(query)}&limit=100`)
```

If there are more than 100 questionnaires using this vocabulary, it probably wouldn't fit anyway. If we select just the questionnaires that do use it, we could even set a limit of 5 and say "... and 20 more others".

Reply:
I can not query just the questionnaires because I need to display both questions and questionnaires names they belong to.